hbase-server/src/main/java/org/apache/hadoop/hbase/master/assignment/RegionStateNode.java

```diff
@@ -206,14 +213,16 @@ public ServerName setRegionLocation(final ServerName serverName) {
   public TransitRegionStateProcedure setProcedure(TransitRegionStateProcedure proc) {
     assert this.procedure == null;
     this.procedure = proc;
-    ritMap.put(regionInfo, this);
+    // Continue to add RITs here. This way we do not need to update a lot of code.
+    addInTransition();
     return proc;
   }
 
   public void unsetProcedure(TransitRegionStateProcedure proc) {
     assert this.procedure == proc;
     this.procedure = null;
-    ritMap.remove(regionInfo, this);
+    // We used to remove RITs from regionStates here. Now TransitRegionStateProcedure does
```

Comment:
If we do not remove RITs when unsetProcedure, it will be strange that we add RITs when setProcedure? Or maybe we can add a new flag called remove RIT for this method, when bypassing procedures through HBCK, we set the flag to false so we will not remove the RIT state?

Reply:
> If we do not remove RITs when unsetProcedure, it will be strange that we add RITs when setProcedure?

The map entry would be updated. This kind of thing is pretty typical for a setter in Java so should not be surprising. 

I left the addition of the region to the RIT map as part of `setProcedure` to minimize the amount of code that must be updated. We could remove RIT map management from `setProcedure` too. But then wherever we have a `setProcedure` we will have an `addRegionInTransition` call just before it. This would be fine! But it would change more code.